api/v1beta1/kafkacluster_types.go

```diff
@@ -303,10 +303,17 @@ type StorageConfig struct {
 
 //ListenersConfig defines the Kafka listener types
 type ListenersConfig struct {
-	ExternalListeners  []ExternalListenerConfig `json:"externalListeners,omitempty"`
-	InternalListeners  []InternalListenerConfig `json:"internalListeners"`
-	SSLSecrets         *SSLSecrets              `json:"sslSecrets,omitempty"`
-	ServiceAnnotations map[string]string        `json:"serviceAnnotations,omitempty"`
+	ExternalListeners []ExternalListenerConfig `json:"externalListeners,omitempty"`
+	InternalListeners []InternalListenerConfig `json:"internalListeners"`
+	// ClientSSLCertSecret is a reference to that secret where custom client SSL certificate can be provided
+	// It can be used by the koperator, cruise control to communicate on SSL with
+	// the internal listener which is used for the interbroker communication.
+	// The included certificate have to be signed by the same CA as the corresponding internal listener's server certificate.
+	// Secret has to contain the keystore and truststore in jks and the password for them in base64 encoded format.
+	// Data fields must be: keystore.jks, truststore.jks, password
+	ClientSSLCertSecret corev1.LocalObjectReference `json:"clientSSLCertSecret,omitempty"`
```

Comment:
Since this is not a listener related config but rather a config for internal clients IMHO it should be moved outside of `ListenersConfig`

Reply:
I think it is related to the listener because this client cert corresponds to the custom server cert which can be set by the user.  I think if we move it out somewhere else, it can mislead the user. Very likely he wouldn't set this also with the custom internal listener

Reply:
if this cert will be used by `koperator` for example than it's a koperator config for me.